main.py

```diff
@@ -54,76 +55,81 @@
-        part += 1
-        texts[i] = f"Chapter {chapter} Part {part}\n{texts[i]}"
-
-# Remove info proceeding the chapter title
-for i in range(len(texts)):
-    firstline = texts[i].split("\n")[0]
-    if "Chapter" in firstline:
-        texts[i] = re.sub(fr"(Arc {arcnumber} Chapter \w+ – [^\n\r]*\n?)(.* ― Complete\n?)", r"\1", texts[i], flags=re.S | re.I)
+    # Every chapter has a title in the first line of the text. However, the parts do not have such a title.
+    # Iterate through the texts and add a title to the first line of each part, incrementing the part number each time until the next time we encounter a chapter title (e.g. "Chapter 2 Part 1").
+    part = 1
+    chapter = "0"
+    for i in range(len(texts)):
+        firstline = texts[i].split("\n")[0]
         
-# Dump the processed text into a file if requested, for debugging purposes
-if args.dump:
-    with open(os.path.join(outputdir, f"Arc {arcnumber} Processed.txt"), "w", encoding="utf-8") as file:
-        file.write("\n\n".join(texts))
-        print("Dumped processed text to file")
-        exit()
-
-def summarize(i: int) -> str:
+        if "Chapter" in firstline and "Part" not in firstline:
+            part = 1
+            chapter = str(re.search(r"(?<=Chapter )\w+", firstline).group(0))
+            
+        if "Chapter" not in firstline:
+            part += 1
+            texts[i] = f"Chapter {chapter} Part {part}\n{texts[i]}"
+
+    # Remove info proceeding the chapter title
+    for i in range(len(texts)):
+        firstline = texts[i].split("\n")[0]
+        if "Chapter" in firstline:
+            texts[i] = re.sub(fr"(Arc {arcnumber} Chapter \w+ – [^\n\r]*\n?)(.* ― Complete\n?)", r"\1", texts[i], flags=re.S | re.I)
+            
+    # Dump the processed text into a file if requested, for debugging purposes
+    if args.dump:
+        with open(os.path.join(outputdir, f"Arc {arcnumber} Processed.txt"), "w", encoding="utf-8") as file:
+            file.write("\n\n".join(texts))
+            print("Dumped processed text to file")
+            exit()
+
+def summarize(text: str, index: int|None = None) -> str:
     """Summarizer process. Not meant to be called directly.
 
     Args:
-        i (int): The index of the part of the chapter to summarize
+        text (str): The text to summarize
+        index (int): The index of the part of the chapter to summarize
     """
-    prompt = "\n<END>\n\nCreate a comprehensive plot synopsis of this part of a chapter from a book.\nMake your plot synopsis as lengthy and detailed as possible."
-    total = texts[i] + prompt
+    if args.custom_text:
+        prompt = "\n<END>\n\nCreate a comprehensive plot synopsis of this passage.\nMake your plot synopsis as lengthy and detailed as possible."
+    else:
+        prompt = "\n<END>\n\nCreate a comprehensive plot synopsis of this part of a chapter from a book.\nMake your plot synopsis as lengthy and detailed as possible."
+    total = text + prompt
```

Comment:
so if we're only going to be chunking the text if it isnt custom, and if it is custom, we just pass everything to chatgpt, we cant complain that its too long if we provide no way to manually get around that.

at the very least make the arg take in multiple files or something. this at least allows the user to manually chunk it into parts

that or let the user specify a custom delimiter

Reply:
we should probably just chunk whether or not it's custom, i just couldnt be bothered lmao.
ill prob do that later

Reply:
> we should probably just chunk whether or not it's custom, i just couldnt be bothered lmao. ill prob do that later

so do you want to add in chunking and let the user specify a custom delimiter, or do you want to allow the user to supply multiple files as input?

either approach works, its your call.

also since a decent amount of our code is sectioned into functions, it shouldnt be hard to implement bulk processing via either approach